packages/components/src/border-control/types.ts

```diff
@@ -84,7 +84,7 @@ export type BorderControlProps = ColorProps &
 		 * A custom CSS class name to be assigned to the border control's
 		 * dropdown popover content.
 		 */
-		popoverContentClassName?: string;
+		__unstablePopoverProps?: any;
```

Comment:
This is indeed a new approach. It definitely feels cleaner! My only hunch is that we're still exposing "popover props" on both components

If we decided to stop using popovers for our UI, suddenly these props wouldn't make sense anymore.

Also, could we look into using the internal context system in order to communicate between `BorderBoxControl` and `BorderControl`, instead of adding a new prop? (an example of this can be see in `Card` [link 1](https://github.com/WordPress/gutenberg/blob/d11bea5abc47acba945bfc129ac406fc7e227b66/packages/components/src/card/card/component.js#L44-L57), [link 2](https://github.com/WordPress/gutenberg/blob/6eff9a514591f3b14f667e94b745eaaa507d912a/packages/components/src/card/card-body/hook.js#L23))

Reply:
> My only hunch is that we're still exposing "popover props" on both components. If we decided to stop using popovers for our UI, suddenly these props wouldn't make sense anymore.

We'd either stick with a UI that is positioned on the side and in which case popoverPlacement and popoverOffset still make sense, or we'd have removed popovers entirely and in which case, we can safely remove the props as there's no need for them anymore anyway.

> Also, could we look into using the internal context system in order to communicate between BorderBoxControl and BorderControl, instead of adding a new prop? (an example of this can be see in Card [link 1](https://github.com/WordPress/gutenberg/blob/d11bea5abc47acba945bfc129ac406fc7e227b66/packages/components/src/card/card/component.js#L44-L57), [link 2](https://github.com/WordPress/gutenberg/blob/6eff9a514591f3b14f667e94b745eaaa507d912a/packages/components/src/card/card-body/hook.js#L23))

Yeah, this is something I thought about too. I can give it a try.

Reply:
This approach is looking good to me. I've closed the alternate approaches to updating `BorderControl` and `BorderBoxControl` popover positioning (https://github.com/WordPress/gutenberg/pull/40874, https://github.com/WordPress/gutenberg/pull/40836)